File: StickyNotes-Mobile-App/app/android/SignInService.kt
import android.util.Log
import okhttp3.*
import org.json.JSONObject
import java.io.IOException

object SignInService {

    private const val TAG = "SignInService"
    private const val URL = "https://StickyNotes.com/signin" // This url is not published to GitHub

    fun signIn(email: String, password: String, callback: (Result<User>) -> Unit) {
        val client = OkHttpClient()

        val requestBody = FormBody.Builder()
            .add("email", email)
            .add("password", password)
            .build()

        val request = Request.Builder()
            .url(URL)
            .post(requestBody)
            .build()

        client.newCall(request).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                Log.e(TAG, "Sign in failed", e)
                callback(Result.failure(e))
            }

            override fun onResponse(call: Call, response: Response) {
                if (!response.isSuccessful) {
                    callback(Result.failure(IOException("Unexpected code $response")))
                    return
                }

                try {
                    val responseBody = response.body?.string()
                    val jsonObject = JSONObject(responseBody ?: "")
                    val user = User(
                        id = jsonObject.getString("id"),
                        name = jsonObject.getString("name"),
                        email = jsonObject.getString("email")
                    )
                    callback(Result.success(user))
                } catch (e: Exception) {
                    callback(Result.failure(e))
                }
            }
        })
    }
}

File: StickyNotes-Mobile-App/app/ios/SignInService.swift
import Foundation

struct SignInService {
    func signIn(email: String, password: String, completion: @escaping (Result<User, Error>) -> Void) {
        // URL for the sign-in API endpoint
        let url = URL(string: "https://StickyNotes.com/signin")! // This url is not published to the public on GitHub

        // Create the request
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
        request.timeoutInterval = 30

        // Create the JSON body
        let body: [String: Any] = ["email": email, "password": password]
        request.httpBody = try? JSONSerialization.data(withJSONObject: body)

        // Perform the request
        URLSession.shared.dataTask(with: request) { data, response, error in
            if let error = error {
                completion(.failure(error))
                return
            }
            guard let data = data else {
                completion(.failure(NSError(domain: "", code: -1, userInfo: nil)))
                return
            }
            // Parse the JSON response
            do {
                let user = try JSONDecoder().decode(User.self, from: data)
                completion(.success(user))
            }
            catch {
                completion(.failure(error))
            }
        }.resume()
    }
}
